odl/tomo/geometry/parallel.py

```diff
@@ -125,84 +171,273 @@ def det_to_src(self, angles, dpar, normalized=True):
+        realize a configuration that cannot be reached by rotation
+        and translation, or if the transformation to be applied is
+        already given as a matrix.
+
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the angle interval.
+        dpart : 1-dim. `RectPartition`
+            Partition of the detector parameter interval.
+        init_matrix : `array_like`, shape ``(2, 2)`` or ``(2, 3)``, optional
+            Transformation matrix whose left ``(2, 2)`` block is multiplied
+            with the default ``det_pos_init`` and ``det_axis_init`` to
+            determine the new vectors. If present, the third column acts
+            as a translation after the initial transformation.
+            The resulting ``det_axis_init`` will be normalized.
+
+        Returns
+        -------
+        geometry : `Parallel2dGeometry`
+            The resulting geometry.
+
+        Examples
+        --------
+        Mirror the second unit vector, creating a left-handed system:
+
+        >>> apart = odl.uniform_partition(0, np.pi, 10)
+        >>> dpart = odl.uniform_partition(-1, 1, 20)
+        >>> matrix = np.array([[1, 0],
+        ...                    [0, -1]])
+        >>> geom = Parallel2dGeometry.frommatrix(apart, dpart, matrix)
+        >>> e_x, e_y = np.eye(2)  # standard unit vectors
+        >>> np.allclose(geom.det_pos_init, -e_y)
+        True
+        >>> np.allclose(geom.det_axis_init, e_x)
+        True
+        >>> np.allclose(geom.translation, (0, 0))
+        True
+
+        Adding a translation with a third matrix column:
+
+        >>> matrix = np.array([[1, 0, 1],
+        ...                    [0, -1, 1]])
+        >>> geom = Parallel2dGeometry.frommatrix(apart, dpart, matrix)
+        >>> np.allclose(geom.translation, (1, 1))
+        True
+        >>> np.allclose(geom.det_pos_init, -e_y + (1, 1))
+        True
+        """
+        # Get transformation and translation parts from `init_matrix`
+        init_matrix = np.asarray(init_matrix, dtype=float)
+        trafo_matrix = init_matrix[:, :2]
+        translation = init_matrix[:, 2:].squeeze()
+
+        # Transform the default vectors
+        default_det_pos_init = cls._default_config['det_pos_init']
+        default_det_axis_init = cls._default_config['det_axis_init']
+        vecs_to_transform = [default_det_axis_init]
+        transformed_vecs = transform_system(
+            default_det_pos_init, None, vecs_to_transform, matrix=trafo_matrix)
+
+        # Use the standard constructor with these vectors
+        det_pos_init, det_axis_init = transformed_vecs
+        if translation.size == 0:
+            kwargs = {}
+        elif translation.size == 2:
+            kwargs = {'translation': translation}
+
+        instance = cls(apart, dpart, det_pos_init,
+                       det_axis_init=det_axis_init, **kwargs)
+        instance.__init_matrix = init_matrix
```

Comment:
Yeah this is pretty much work in progress, the repr is also wrong currently since it doesn't print the `frommatrix` thing in the matrix init case.
I would, though, really want to use the baseline init pattern in repr. And in fact, there is nothing you cannot do that can be done with a matrix, so I'll just remove this property and simplify repr.